network_utils/src/network.rs

```diff
@@ -44,6 +50,54 @@ impl NetworkClient {
         }
     }
 
+    pub async fn send_recv_bytes_stream(
+        &self, buf: Vec<u8>,
+        rx_cancellation: oneshotReceiver<()>
+    ) -> Result<InflightStream, SuiError> {
+        let result = self
+            .send_recv_bytes_stream_internal(buf, rx_cancellation)
+            .await;
+        match result {
+            Ok(r) => Ok(r),
+            Err(error) => Err(SuiError::ClientIoError {
+                error: format!("{}", error),
+            }),
+        }
+    }
+
+    async fn send_recv_bytes_stream_internal(
+        &self, buf: Vec<u8>,
+        rx_cancellation: oneshotReceiver<()>
+    ) -> Result<InflightStream, io::Error> {
+        let address = format!("{}:{}", self.base_address, self.base_port);
+        let mut stream = connect(address, self.buffer_size).await?;
+        // Send message
+        time::timeout(self.send_timeout, stream.write_data(&buf)).await??;
+        let (tx_output, tr_output) = channel(100);
+        let join_handle = tokio::spawn(async move {
```

Comment:
Here we spawn a tokio task. Tasks are handy for processes that need to drive themselves forward, ie need polling. But this stream will be polled by the handle_batch_streaming function. Does it need to be a task?

Similarly, cancel here is handy if this is a task that has a schedule of its own. If it is just a stream, then we can just drop the stream to close it.